rdt/transformers/pii/anonymizer.py

```diff
@@ -209,3 +214,148 @@ def __repr__(self):
+        ``function``.
+
+        Args:
+            data (pandas.Series):
+                Data to fit the transformer to.
+        """
+        super()._fit(columns_data)
+        unique_values = columns_data[columns_data.notna()].unique()
+        unique_data_length = len(unique_values)
+        generated_values = [self._function() for _ in range(unique_data_length)]
+        if len(set(generated_values)) != unique_data_length:
+            counter = 0
+            while len(set(generated_values)) != unique_data_length:
+                generated_values.append(self._function())
+                counter += 1
+                if counter == 10:
+                    error_msg = (
+                        'The Faker function you specified is only able to generate '
+                        f'{len(set(generated_values))} unique values, which is not enough to '
+                        'create a mapping. Please use a different Faker function for this column.'
+                    )
+                    raise ValueError(error_msg)
+
+        generated_values = list(set(generated_values))
+        self._mapping_dict = dict(zip(unique_values, generated_values))
+        self._reverse_mapping_dict = dict(zip(generated_values, unique_values))
+
+    def _transform(self, columns_data):
+        """Replace each category with a numerical representation.
+
+        Map the input ``columns_data`` using the previously generated values for each one, then
+        use a ``LabelEncoder`` with ``add_noise=True`` to generate the numerical representation
+        for each one. If the  ``columns_data`` contain unknown values, a ``ValueError`` will be
+        raised with the unknown categories.
+
+        Args:
+            data (pandas.Series):
+                Data to transform.
+
+        Returns:
+            pd.Series
+        """
+        unique_values = columns_data[columns_data.notna()].unique()
+        new_values = list(set(unique_values) - set(self._mapping_dict))
+        if new_values:
+            new_values = [str(value) for value in new_values]
+            if len(new_values) < 5:
+                new_values = ', '.join(new_values)
+                error_msg = (
+                    'The data you are transforming has new, unexpected values '
+                    f'({new_values}). Please fit the transformer again using this '
+                    'new data.'
+                )
+            else:
+                diff = len(new_values) - 5
+                new_values = ', '.join(new_values[:5])
+                error_msg = (
+                    'The data you are transforming has new, unexpected values '
+                    f'({new_values} and {diff} more). Please fit the transformer again '
+                    'using this new data.'
+                )
+
+            raise ValueError(error_msg)
+
+        mapped_data = pd.DataFrame(columns_data.map(self._mapping_dict))
+        self._label_encoder = LabelEncoder(add_noise=True)
```

Comment:
Instead of putting the `LabelEncoder` in here, can we set the `NEXT_TRANSFORMERS` attribute? This was meant to be a way for one transformer to set the next transformer to use for any of its outputs